scripts/generate_image.ps1

```diff
@@ -0,0 +1,175 @@
+$BUILD_ROOT="${GOPATH}\src\github.com\aws\amazon-ecs-agent"
+
+if(!$(Test-Path -Path "${WORK_ROOT}")) {
+  New-Item -Path "${WORK_ROOT}" -Type directory
+}
+
+echo "==============================================================================="
+echo "Building windows images for commit ${COMMIT_SHA}"
+echo "==============================================================================="
+
+# Install Git
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+echo "Downloading Git..."
+Invoke-WebRequest "https://github.com/git-for-windows/git/releases/download/v${GitVersion}.windows.1/Git-${GitVersion}-64-bit.exe" -OutFile "${env:TEMP}\git-installer.exe" -UseBasicParsing
+echo "Installing Git..."
+Start-Process $env:TEMP\git-installer.exe -ArgumentList '/SILENT /COMPONENTS="icons,ext,ext\reg\shellhere,assoc,assoc_sh"' -NoNewWindow -Wait
+$env:Path = ${env:Path} + ";${env:ProgramFiles}/Git/cmd"
+[Environment]::SetEnvironmentVariable("Path", $env:Path + ";${env:ProgramFiles}/Git/cmd", [EnvironmentVariableTarget]::Machine)
+git config --global user.name "nobody"
+git config --global user.email "nobody@example.localhost"
+
+# Install Go
+echo "Downloading Go..."
+Invoke-WebRequest "https://storage.googleapis.com/golang/go${GoVersion}.windows-amd64.msi" -OutFile "${env:TEMP}\go-installer.msi" -UseBasicParsing
+echo "Installing Go..."
+Start-Process "msiexec.exe" -ArgumentList "/quiet /i ${env:TEMP}\go-installer.msi" -Wait
+New-Item -ItemType directory -Path "${WORK_ROOT}\go\bin"
+New-Item -ItemType directory -Path "${WORK_ROOT}\go\src"
+[Environment]::SetEnvironmentVariable("Path", $env:Path + ";C:\Go\bin;${WORK_ROOT}\go\bin", [EnvironmentVariableTarget]::Machine)
+[Environment]::SetEnvironmentVariable("GOPATH", "${WORK_ROOT}\go", [EnvironmentVariableTarget]::Machine)
+$env:Path=$env:Path + ";C:\Go\bin;${WORK_ROOT}\go\bin"
+$env:GOPATH="${WORK_ROOT}\go"
+
+echo "Checking out ${GIT_REPO} at ${COMMIT_SHA}"
+git clone "${GIT_REPO}" "${BUILD_ROOT}" 2>&1 | % {$_.ToString()}
+cd "${BUILD_ROOT}"
+git config --add remote.origin.fetch '+refs/pull/*/head:refs/remotes/origin/pr/*'
+git fetch 2>&1 | % {$_.ToString()}
+git checkout "${COMMIT_SHA}" 2>&1 | % {$_.ToString()}
+
+echo "Starting Docker..."
+Start-Service -Name 'docker'
+
+echo "Waiting for Docker to start..."
+$dockerSrv = Get-Service -Name 'docker'
+$stat = $dockerSrv.WaitForStatus('Running', '00:15:00')
+
+echo "Waiting for Docker to become responsive..."
+docker ps
+if (${LastExitCode} -ne 0) {
+  echo "Docker ps didn't go well." -logLevel "ERROR"
+  echo $_.Exception.Message -logLevel "ERROR"
+}
+
+echo "-------------------------------------------------------------------------------"
+echo "Generating Windows Images..."
+echo "-------------------------------------------------------------------------------"
+
+$ErrorActionPreference="Stop"
+
+$application = Get-WmiObject Win32_Product -filter "Name='AWS Tools for Windows'"
+if ($application) {
+    $application.Uninstall()
```

Comment:
Why are we uninstalling the AWS Tools here? This operation has been known to fail in the past

Reply:
If the old version isn't been uninstalled, the new version does not get referenced.